File: app/iosApp/iosApp/UI/Components/ErrorView.swift
import SwiftUI
import SharedLogic

struct ErrorView: View {
    let type: ErrorType
    let designSystem: DesignSystem
    let onRetry: () -> Void

    var body: some View {
        let titleToken = getTitleToken()
        let msgToken = getMsgToken()
        let emoji = getEmoji()

        ZStack {
            VStack(spacing: 24) {
                Text(emoji)
                    .font(.system(size: 64))

                VStack(spacing: 8) {
                    Text(designSystem.string(token: titleToken))
                        .font(.title2)
                        .bold()
                        .foregroundColor(designSystem.color(.textPrimary))

                    Text(designSystem.string(token: msgToken))
                        .font(.body)
                        .multilineTextAlignment(.center)
                        .foregroundColor(designSystem.color(.textSecondary))
                        .padding(.horizontal, 32)
                }

                Button(action: onRetry) {
                    Text(designSystem.string(token: .errorRetry))
                        .fontWeight(.bold)
                        .padding(.horizontal, 32)
                        .padding(.vertical, 12)
                        .background(designSystem.color(.accentEnergy))
                        .foregroundColor(designSystem.color(.textInverted))
                        .cornerRadius(12)
                }
            }
            .padding(.vertical, 40)
            .frame(maxWidth: .infinity)
            .background(designSystem.color(.glassBase).opacity(0.3))
            .background(.ultraThinMaterial)
            .cornerRadius(28)
            .padding(24)
        }
        .frame(maxHeight: .infinity)
    }

    private func getEmoji() -> String {
        switch type {
        case .noInternet: return "🌐"
        case .serverUnavailable: return "📡"
        default: return "⚠️"
        }
    }

    private func getTitleToken() -> StringToken {
        switch type {
        case .noInternet: return .errorNoInternetTitle
        case .serverUnavailable: return .errorServerTitle
        default: return .errorUnknownTitle
        }
    }

    private func getMsgToken() -> StringToken {
        switch type {
        case .noInternet: return .errorNoInternetMsg
        case .serverUnavailable: return .errorServerMsg
        default: return .errorUnknownMsg
        }
    }
}

File: app/androidApp/src/main/kotlin/application/liedetector/ui/components/state/ErrorView.kt
package application.liedetector.ui.components.state

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import application.liedetector.theme.utils.composeColor
import application.liedetector.uicore.theme.*
import application.liedetector.uicore.theme.tokens.ColorToken
import application.liedetector.uicore.theme.tokens.DimenToken
import application.liedetector.uicore.theme.tokens.StringToken
import application.liedetector.uicore.types.ErrorType

/**
 * Renders an error state as a full-page overlay within the screen's content area.
 */
@Composable
fun ErrorView(
    type: ErrorType,
    onRetry: () -> Unit,
    modifier: Modifier = Modifier
) {
    val designSystem = LocalDesignSystem.current

    val titleToken = when (type) {
        ErrorType.NO_INTERNET -> StringToken.ERROR_NO_INTERNET_TITLE
        ErrorType.SERVER_UNAVAILABLE -> StringToken.ERROR_SERVER_TITLE
        else -> StringToken.ERROR_UNKNOWN_TITLE
    }

    val msgToken = when (type) {
        ErrorType.NO_INTERNET -> StringToken.ERROR_NO_INTERNET_MSG
        ErrorType.SERVER_UNAVAILABLE -> StringToken.ERROR_SERVER_MSG
        else -> StringToken.ERROR_UNKNOWN_MSG
    }

    Box(
        modifier = modifier
            .fillMaxSize()
            .background(designSystem.composeColor(ColorToken.BACKGROUND).copy(alpha = 0.8f))
            .padding(designSystem.dimen(DimenToken.PADDING_ERROR).dp),
        contentAlignment = Alignment.Center
    ) {
        Column(horizontalAlignment = Alignment.CenterHorizontally) {
            Text(
                text = designSystem.string(titleToken),
                style = MaterialTheme.typography.headlineMedium,
                color = designSystem.composeColor(ColorToken.TEXT_PRIMARY),
                textAlign = TextAlign.Center
            )
            Spacer(modifier = Modifier.height(designSystem.dimen(DimenToken.SPACING_SMALL).dp))
            Text(
                text = designSystem.string(msgToken),
                style = MaterialTheme.typography.bodyLarge,
                color = designSystem.composeColor(ColorToken.TEXT_SECONDARY),
                textAlign = TextAlign.Center
            )
            Spacer(modifier = Modifier.height(designSystem.dimen(DimenToken.SPACING_LARGE).dp))
            Button(
                onClick = onRetry,
                colors = ButtonDefaults.buttonColors(
                    containerColor = designSystem.composeColor(ColorToken.ACCENT_ENERGY),
                    contentColor = designSystem.composeColor(ColorToken.TEXT_INVERTED)
                )
            ) {
                Text(designSystem.string(StringToken.ERROR_RETRY))
            }
        }
    }
}
